astropy/units/equivalencies.py

```diff
@@ -561,6 +564,268 @@ def mass_energy():
+        """
+        Convert irreducible quantities. Unit powers are unity.
+        E.g. meter to second, but not meter-squared to second-squared
+
+        Parameters
+        ----------
+        source_quantity: astropy.units.Quantity
+            source quantity
+        target_unit: astropy.units.IrreducibleUnit
+            target quantity
+        """
+        source_quantity_decomposed = source_quantity.decompose()
+        source_quantity_unit = source_quantity_decomposed.unit
+        conversion_factor = self._base_conversion(
+            source_quantity_unit, target_unit
+        ).value
+        return source_quantity_decomposed.value * conversion_factor
+
+    def _homogeneous_conversion(
+            self, source_quantity,
+            target_unit):
+        """Composite quantity conversion. E.g. meter-cubed to second-cubed,
+        but not m/s to dimensionless units.
+
+        Parameters
+        ----------
+        source_quantity_value: float
+            source quantity value
+        source_units: list
+            list of units
+
+        """
+        assert len(source_quantity.unit.bases) == len(target_unit.bases) == 1
+        assert source_quantity.unit.powers == target_unit.powers
+
+        res = source_quantity.value
+
+        res *= self._irreducible_conversion(
+            1.0 * source_quantity.unit.bases[0],
+            target_unit.bases[0]
+        )**target_unit.powers[0]
+
+        return res
+
+
+def geometrized(physical_type):
+    """
+    Convert quantities to geometric units.
+
+    Parameters
+    ----------
+    physical_type: str or `~astropy.units.PhysicalType`
+        Physical type to convert.
+        Argument to :meth:`astropy.units.get_physical_type`
+
+    Examples
+    --------
+    >>> from astropy import constants as c, units as u
+    >>> mass = 2 * u.M_sun
+    >>> mass.to('km', equivalencies=u.geometrized('mass'))
+    <Quantity 2.95325008 km>
+    >>> time = 1 * u.microsecond
+    >>> time.to('solMass', equivalencies=u.geometrized('time'))
+    <Quantity 0.20302545 solMass>
+    >>> time.to('kg', equivalencies=u.geometrized('time'))
+    <Quantity 4.03697802e+29 kg>
+    """
+    physical_type = physical.get_physical_type(physical_type)
+    mass_length_time_physical_types = [
+        physical.length,
+        physical.mass,
+        physical.time,
+    ]
+
+    volume_physical_types = [
+        physical.volume
+    ]
+
+    area_physical_types = [
+        physical.area,
+    ]
+
+    temperature_physical_types = [
+        physical.temperature,
+    ]
+
+    velocity_physical_types = [
+        physical.velocity,
+    ]
+
+    allowed_physical_types = mass_length_time_physical_types
+    allowed_physical_types += volume_physical_types
+    allowed_physical_types += area_physical_types
+    allowed_physical_types += temperature_physical_types
+    allowed_physical_types += velocity_physical_types
+
+    assert physical_type in allowed_physical_types, (
```

Comment:
Assert statements are stripped in certain python compilation modes. If you don't want this check run in "optimized mode", this is fine, otherwise use an `if` statement and manually raise the desired error

Reply:
I wanted to take this opportunity to check if this is done in the CI pipeline.